main.go

```diff
@@ -0,0 +1,393 @@
+	organization               = flag.String("organization", "", "The organizationID that is the subject of this audit")
+	subject                    = flag.String("subject", "", "The admin user to for the organization that can use the Directory API to list users")
+	cx                         = flag.String("cx", "", "Workspace Customer ID number")
+	delay                      = flag.Int("delay", 2*1000, "delay in ms for each user iterated")
+	allProjects                = make(map[string]*assetpb.ResourceSearchResult)
+	allOrganizations           = make(map[string]*assetpb.ResourceSearchResult)
+	limiter                    *rate.Limiter
+	svcAccountJSONBytes        []byte
+)
+
+const (
+	maxRequestsPerSecond float64 = 3 // "golang.org/x/time/rate" limiter to throttle operations
+	burst                int     = 1
+	maxPageSize          int64   = 1000
+)
+
+func getOrganizations(ctx context.Context, u admin.User) ([]*cloudresourcemanager.Organization, error) {
+	glog.V(50).Infof("             Getting Organizations for user %s", u.PrimaryEmail)
+	defer wg.Done()
+
+	var crmService *cloudresourcemanager.Service
+
+	if *impersonatedServiceAccount != "" {
+		ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: *impersonatedServiceAccount,
+			Scopes:          []string{cloudresourcemanager.CloudPlatformReadOnlyScope},
+			Subject:         u.PrimaryEmail,
+		})
+		if err != nil {
+			glog.Errorf("Error creating CRM credentials for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Organization{}, err
+		}
+		crmService, err = cloudresourcemanager.NewService(ctx, option.WithTokenSource(ts))
+		if err != nil {
+			glog.Errorf("Error creating CRM Service for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Organization{}, err
+		}
+	} else {
+		cred, err := google.CredentialsFromJSONWithParams(ctx, svcAccountJSONBytes, google.CredentialsParams{
+			Scopes:  []string{cloudresourcemanager.CloudPlatformReadOnlyScope},
+			Subject: u.PrimaryEmail,
+		})
+		if err != nil {
+			glog.Errorf("Error creating CRM credentials for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Organization{}, err
+		}
+		crmService, err = cloudresourcemanager.NewService(ctx, option.WithCredentials(cred))
+		if err != nil {
+			glog.Errorf("Error creating CRM Service for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Organization{}, err
+		}
+	}
+
+	organizations := make([]*cloudresourcemanager.Organization, 0)
+
+	// do not apply a  filter here; we need to audit all projects
+	//  https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v1#SearchOrganizationsRequest
+	noFilter := ""
+
+	req := crmService.Organizations.Search(&cloudresourcemanager.SearchOrganizationsRequest{Filter: noFilter, PageSize: maxPageSize})
+
+	if err := req.Pages(ctx, func(page *cloudresourcemanager.SearchOrganizationsResponse) error {
```

Comment:
Do not split your if statements into new lines, see the other comment below about this.